packages/ui/src/components/TextInputV2/index.tsx

```diff
@@ -0,0 +1,374 @@
+`
+
+const StateStack = styled(Stack)`
+  padding: ${({ theme }) => `0 ${theme.space['2']}`};
+`
+
+const BasicSuffixStack = styled(Stack)`
+  padding: ${({ theme }) => `0 ${theme.space['2']}`};
+  border-left: 1px solid;
+  border-color: inherit;
+`
+
+const CTASuffixStack = styled(Stack)`
+  padding: ${({ theme }) => `0 ${theme.space['1']}`};
+  border-left: 1px solid;
+  border-color: inherit;
+`
+
+type StyledInputProps = {
+  isSuccess: boolean
+  isError: boolean
+  isClearable: boolean
+}
+const StyledInput = styled('input', {
+  shouldForwardProp: prop =>
+    !['isSuccess', 'isError', 'isClearable'].includes(prop),
+})<StyledInputProps>`
+  flex: 1;
+  border: none;
+  outline: none;
+  height: 100%;
+  padding-left: ${({ theme }) => theme.space['2']};
+  background: transparent;
+`
+
+type StyledInputWrapperProps = {
+  hasFocus: boolean
+  size: TextInputSize
+}
+const StyledInputWrapper = styled('div', {
+  shouldForwardProp: prop => !['hasFocus', 'size'].includes(prop),
+})<StyledInputWrapperProps>`
+  display: flex;
+  flex-direction: row;
+  height: ${({ size }) => TEXTINPUT_SIZE_HEIGHT[size]}px;
+
+  background: ${({ theme }) => theme.colors.neutral.background};
+  border: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  & > ${StyledInput} {
+    color: ${({ theme }) => theme.colors.neutral.text};
+    &::placeholder {
+      color: ${({ theme }) => theme.colors.neutral.textWeak};
+    }
+  }
+
+  &[data-success='true'] {
+    border-color: ${({ theme }) => theme.colors.success.border};
+  }
+
+  &[data-error='true'] {
+    border-color: ${({ theme }) => theme.colors.danger.border};
+  }
+
+  &[data-readOnly='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+  }
+
+  &[data-disabled='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+
+    & > ${StyledInput} {
+      color: ${({ theme }) => theme.colors.neutral.textDisabled};
+      &::placeholder {
+        color: ${({ theme }) => theme.colors.neutral.textWeakDisabled};
+      }
+    }
+  }
+
+  &:not([data-disabled='true']):hover {
+    border-color: ${({ theme }) => theme.colors.primary.border};
+  }
+
+  ${({ theme, hasFocus }) =>
+    hasFocus
+      ? `
+  box-shadow: ${theme.shadows.focusPrimary};
+  border: 1px solid ${theme.colors.primary.border};    
+`
+      : null};
+`
+
+type TextInputProps = {
+  autoFocus?: boolean
+  className?: string
+  clearable?: boolean
+  'data-testid'?: string
+  disabled?: boolean
+  error?: string
+  helper?: ReactNode
+  iconName?: ComponentProps<typeof Icon>['name']
+  id?: string
+  label: string
+  labelDescription?: ReactNode
+  loading?: boolean
+  minLength?: number
+  maxLength?: number
+  name?: string
+  onBlur?: DOMAttributes<HTMLInputElement>['onBlur']
```

Comment:
```suggestion
  onBlur?: DOMAttributes<HTMLTextAreaElement>['onBlur']
```

?

Reply:
@DorianMaliszewski text input is indeed an HTML input element, we have a TextArea component which is different. TextInput doesn't support text area 😄